client/app/project/project.controller.js

```diff
@@ -330,6 +331,23 @@
         };
 
         /**
+         * Undo task status change
+         */
+        vm.undo = function(){
+            vm.taskUndoError = false;
+            var projectId = vm.projectData.projectId;
+            var taskId = vm.selectedTaskData.taskId;
+            var resultsPromise = taskService.undo(projectId, taskId);
+            resultsPromise.then(function (data) {
+                setUpSelectedTask(data);
```

Comment:
Should maybe put these lines in before line 342.  Not sure if it's necessary, but thinki the pattern is used elsewhere:
 
vm.resetErrors();
vm.resetStatusFlags();
vm.resetTaskData();